packages/flutter_test/lib/src/all_elements.dart

```diff
@@ -42,20 +71,24 @@ class _DepthFirstChildIterator implements Iterator<Element> {
       return false;
 
     _current = _stack.removeLast();
-    // Stack children in reverse order to traverse first branch first
-    _stack.addAll(_reverseChildrenOf(_current, skipOffstage));
+    _fillChildren(_current);
 
     return true;
   }
 
-  static Iterable<Element> _reverseChildrenOf(Element element, bool skipOffstage) {
+  void _fillChildren(Element element) {
     assert(element != null);
-    final List<Element> children = <Element>[];
+    // If we did not have to follow LTR order and could instead use RTL,
+    // we could avoid reversing this and the operation would be measurably
+    // faster. Unfortunately, a lot of tests depend on LTR order.
+    final ListQueue<Element> reversed = ListQueue<Element>();
     if (skipOffstage) {
-      element.debugVisitOnstageChildren(children.add);
+      element.debugVisitOnstageChildren(reversed.addFirst);
```

Comment:
Could this just be addLast and then removeLast below? Then this would just be a regular list. How is ListQueue over list beneficial here?